lte/gateway/c/session_manager/UpfMsgManageHandler.cpp

```diff
@@ -123,4 +130,66 @@ void UpfMsgManageHandler::SetUPFSessionsConfig(
   return;
 }
 
+void UpfMsgManageHandler::SendPagingRequest(
+    ServerContext* context, const UPFPagingInfo* page_request,
+    std::function<void(Status, SmContextVoid)> response_callback) {
+  auto& pag_req = *page_request;
+
+  uint32_t fte_id     = pag_req.local_f_teid();
+  std::string ip_addr = pag_req.ue_ip_addr();
+  struct in_addr ue_ip;
+  IPAddress req = IPAddress();
+
+  inet_aton(ip_addr.c_str(), &ue_ip);
+  req.set_version(IPAddress::IPV4);
+  req.set_address(&ue_ip, sizeof(struct in_addr));
+
+  mobilityd_client_->get_subscriberid_from_ipv4(
+      req, [this, fte_id, response_callback](
+               Status status, const SubscriberID& sid) {
+        if (!status.ok()) {
+          MLOG(MERROR) << "Subscriber could not be found for ip ";
+        }
+        const std::string& imsi = sid.id();
+        conv_enforcer_->get_event_base().runInEventBaseThread(
+            [this, imsi, fte_id, response_callback]() {
+              if (!imsi.length()) {
+                MLOG(MERROR) << "Subscriber is NULL";
+                Status status(grpc::NOT_FOUND, "Sesion Not found");
+                response_callback(status, SmContextVoid());
+                return;
+              }
+
+              // retreive session_map entry
+              auto session_map = session_store_.read_sessions({imsi});
+              /* Search with session search criteria of IMSI and session_id and
+               * find  respective sesion to operate
+               */
+              SessionSearchCriteria criteria(imsi, IMSI_AND_TEID, fte_id);
+
+              auto session_it =
+                  session_store_.find_session(session_map, criteria);
+              if (!session_it) {
+                MLOG(MERROR) << "No session found in SessionMap for IMSI "
+                             << imsi << " with teid " << fte_id;
+                Status status(grpc::NOT_FOUND, "Sesion Not found");
```

Comment:
Sesion typo -> `session`

Specify in the message that the session was not found for IMSI/Teid pair.

Reply:
Yes. Corrected typo and changed status message.